custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py

```diff
@@ -177,6 +180,30 @@ def _predict(self, **kwargs):
         out_df = pd.read_csv(StringIO(out_csv))
         return out_df
 
+    def predict_unstructured(self, data, **kwargs):
+        mimetype = kwargs.get(UnstructuredDtoKeys.MIMETYPE, "")
+        query = kwargs.get(UnstructuredDtoKeys.QUERY, "")
+        charset = kwargs.get(UnstructuredDtoKeys.CHARSET, "utf-8")
+        if isinstance(data, bytes):
+            ba = bytearray(data)
+        elif isinstance(data, str):
+            ba = bytearray(data.encode(charset))
+        else:
+            raise DrumCommonException("data of type {type(data)} is not supported")
+        if isinstance(query, dict):
+            query_dict = query
+        elif isinstance(query, ImmutableMultiDict):
+            query_dict = query.to_dict()
+        jmap = MapConverter().convert(query_dict, self._gateway._gateway_client)
```

Comment:
what is the result of this if `query` ends up being empty string?

Reply:
in generic_predictor.py -> the _materialize method parses the query to a list of name, value pairs and then is converted to dictionary, specifically at line 80

`query_params = dict(urllib.parse.parse_qsl(self._params.get("query_params")))` 

so if initial query being passed to DRUM is empty string, it should be converted to an empty dict() at some point.  Then `query` will only every be a dict or ImmutableMultiDict (i couldn't trace where this happens, but I know that it does in fact happen since `MapConverter().convert` could not convert it to java Map, even though it is a subclass of dictionary.  

Regardless, of this, a few lines up i have 

`query = kwargs.get(UnstructuredDtoKeys.QUERY, "")`

which I will change to 

`query = kwargs.get(UnstructuredDtoKeys.QUERY, dict())`